spec/predicates/deployment.md

```diff
@@ -0,0 +1,477 @@
+A scope has a type and a value. A type ends with its version encoded with `/version`, such as `/v1`. Examples of explicit scope types include Kubernetes's objects such as a pod's cluster ID or a GCP service account. Examples of implicit scope types incude [Google Cloud Binauthz](https://cloud.google.com/binary-authorization/) policy URIs.
+Let's see some examples:
+
+- If we want to "restrict an image to run only on GKE cluster X", the scope type is a "Kubernetes cluster" and "X" is the scope value.
+- If we want to "restrict an image to run only under service account Y", the scope type is "service account" and the scope value is "Y".
+- If we want to "run an image only if it has been vuln scanned", the scope is empty in the sense that the image is allowed to run in any target environment.
+
+### Parsing Rules
+
+This predicate follows the in-toto attestation [parsing rules](https://github.com/in-toto/attestation/blob/main/spec/v1/README.md#parsing-rules).
+Summary:
+
+- Consumers MUST ignore the `decisionDetails` field during verification. The field is purely informational and is intended only
+for troubleshooting and logging.
+- Consumers MUST reject attestations with scope types they do not recognize.
+- The `predicateType` URI includes the major version number and will always change whenever there is a backwards incompatible change.
+- Minor version changes are always backwards compatible and "monotonic". Such changes do not update the `predicateType`.
+- Producers MAY add custom scope types to the `scopes` field. To avoid type name collisions, a scope type MUST be an URI. See [custom scopes](#custom-scopes).
+
+### Verification
+
+#### Configuration
+Verification of a deployment attestation is typically performed by an admission controller prior to deploying an artifact / container.
+The verification configuration MUST be done out-of-band and contain the following pieces of information:
+
+1. Required: The "trusted roots". A trusted root defines an entity that is trusted to generate attestations. A trusted root MUST be configured with the following pieces of information:
+    - Required: The unique identity of the attestation generator. The identity may be a cryptographic public key, an identity in an x509 certificate, etc.
+    - Required: Which scope types the attestation generator is authoritative for.
+2. Optional: Required scopes, which is a set of mandatory scope types that MUST be non-empty for verification to pass. Images MUST have attestation(s) over each scope type in the set in order to be admitted. Required scopes are necessary in an attestation, but not sufficient; other scopes present in the attestation MUST match the current environment in order for it to be considered valid.
```

Comment:
After more thought, I think required scopes type and required scope values should be bound to a trusted root instead of the environment (e.g. cluster) as a whole. We refer to Binauthz policies being bound to the cluster, but that's because we have only a single implicitly trusted root (the binauthz policy evaluator). If we allowed users to bring their own trusted root, it wouldn't make sense to still require a Binauthz policy uri scope.

Reply:
Done, Thanks